extension/src/popup/components/hardwareConnect/LedgerSign/index.tsx

```diff
@@ -80,12 +85,18 @@ export const LedgerSign = () => {
       );
       if (signWithLedger.fulfilled.match(res)) {
         if (shouldSubmit) {
-          dispatch(
+          const submitResp = await dispatch(
             submitFreighterTransaction({
               signedXDR: res.payload,
               networkDetails,
             }),
           );
+          if (
+            submitFreighterTransaction.fulfilled.match(submitResp) &&
+            !isSwap
+          ) {
+            dispatch(addRecentAddress({ publicKey: destination }));
```

Comment:
save recent address for hardware wallet successful send
https://stellarorg.atlassian.net/browse/WAL-385